docs/plugin_autodoc.py

```diff
@@ -0,0 +1,176 @@
+
+    return [
+        get_plugin_info(p, m, base_path)
+        for (p, m) in zip(plugins, plugin_metadata)
+        if "status" not in m and "code" not in m
+    ]
+
+
+def write_index(doc, plugins):
+    doc.write(":::{list-table} Plugin Overview\n")
+    doc.write(":header-rows: 1\n")
+    doc.write(":width: 100%\n")
+    doc.write(":widths: 30 10 30\n\n")
+    doc.write("* - Plugin\n  - Type\n  - Tags\n")
+    sep = "\n\n    "
+    for p in sorted(plugins, key=lambda p: p["name"]):
+        doc.write(f"* - [{p['name']} (@{p['version']})](#{p['id']})\n\n")
+        doc.write(f"    {p['identifier']}\n")
+        doc.write(f"  - {p['type']}\n")
+        doc.write(f"  - {sep.join(p['tags'])}\n")
+    doc.write("\n:::\n\n")
+
+
+def write_merged_data(doc, plugins):
+    tags = set()
+    input_formats = set()
+    output_formats = set()
+    input_datatypes = set()
+    output_datatypes = set()
+
+    for p in plugins:
+        tags.update(p["tags"])
+        for input_def in p["input"]:
+            input_formats.update(input_def["contentType"])
+            input_datatypes.add(input_def["dataType"])
+        for output_def in p["output"]:
+            output_formats.update(output_def["contentType"])
+            output_datatypes.add(output_def["dataType"])
+
+    doc.write("## Overview\n\n")
+    doc.write(f"**Used tags:** {', '.join(f'`{t}`' for t in sorted(tags))}\n\n")
+    doc.write(
+        f"**Input formats:** {', '.join(f'`{f}`' for f in sorted(input_formats))}\\\n"
+    )
+    doc.write(
+        f"**Output formats:** {', '.join(f'`{f}`' for f in sorted(output_formats))}\n\n"
+    )
+    doc.write(
+        f"**Input datatypes:** {', '.join(f'`{f}`' for f in sorted(input_datatypes))}\\\n"
+    )
+    doc.write(
+        f"**Output datatypes:** {', '.join(f'`{f}`' for f in sorted(output_datatypes))}\n\n"
+    )
+
+
+def write_plugin(doc, p):
+    doc.write(f"({p['id']})=\n")
+    doc.write(f"### {p['name']} (@{p['version']})\n\n")
+    doc.write(f"{p['type']} – {', '.join(p['tags'])}\\\n")
+    doc.write(f"*Path:* {{file}}`{p['path']}`\n\n")
+    doc.write(p["description"])
+    doc.write("\n\n")
+    if p["input"]:
+        doc.write("**Inputs:**\n\n")
+        doc.write("| Data Type | Content Type | Required |\n")
+        doc.write("|-----------|--------------| :------: |\n")
+        for data_input in p["input"]:
+            doc.write(
+                f"|{data_input['dataType']}|{', '.join(data_input['contentType'])}|{'✓' if data_input['required'] else '🗙'}|\n"
+            )
+        doc.write("\n\n")
+    if p["output"]:
+        doc.write("**Outputs:**\n\n")
+        doc.write("| Data Type | Content Type | Always |\n")
+        doc.write("|-----------|--------------| :----: |\n")
+        for data_output in p["output"]:
+            doc.write(
+                f"|{data_output['dataType']}|{', '.join(data_output['contentType'])}|{'✓' if data_output['required'] else '🗙'}|\n"
```

Comment:
'🗙' is rendered as an unknown character in Firefox, Chrome, PyCharm etc. on MacOS.

Reply:
Since this is a Unicode character, it depends on what font is installed (and used) to display this character. Can you suggest an alternative that renders nicely on macOS? (in the worst case we could use 'X' or even 'yes' 'no'...)